### PR TITLE
Change on compute_[blob_]kzg_proof function signatures

### PR DESCRIPTION
**Pull request by ppopth**
This PR implements the spec change specified in:

* https://github.com/ethereum/consensus-specs/pull/3280

In addition, this PR also updates the tests in `/tests` and implements the Python binding change with respect to the spec change.

### CODE REVIEW

File: src/test_c_kzg_4844.c
Comment (asn-d6) [inline src/test_c_kzg_4844.c:1093]: Perhaps another approach here would be to evaluate the blob at `input_value` using `evaluate_polynomial_in_evaluation_form` and make sure the output matches what compute_kzg_proof returned.

File: src/c_kzg_4844.c
Comment (asn-d6) [inline src/c_kzg_4844.c:1102]: Hmm, this seems a bit non-standard. Why do we need both `y` and `y_out`?

If we need both, I think it's more standard to use `y` throughout the function and then in the end do `*y_out = y`.

File: bindings/python/ckzg.c
Comment (jtraglia) [inline bindings/python/ckzg.c:91]: Nice catch with the name 🥇

File: src/c_kzg_4844.c
Comment (jtraglia) [inline src/c_kzg_4844.c:1085]: Another nice catch. These lines may be able to be joined into one now btw.

File: tests/verify_kzg_proof/small/verify_kzg_proof_case_commitment_not_in_G1/data.yaml
Comment (asn-d6) [inline tests/verify_kzg_proof/small/verify_kzg_proof_case_commitment_not_in_G1/data.yaml:2]: Were these changes done by running the new test gen from https://github.com/ethereum/consensus-specs/pull/3280/files ? Just wanted to make sure.

**Review by jtraglia — CHANGES_REQUESTED** [review]
Very nice work. Looks great, just one nit.

Also, please run `make format` to fix the formatting complaints. This will actually require a little manual editing afterwards. Sadly, clang-format doesn't work that great with comments. You will need to indent the doxygen parameter doc that's too long & manually join those two lines I mention in the comment.

File: src/c_kzg_4844.c
Comment (jtraglia) [reply src/c_kzg_4844.c:1102]: I agree this a little weird. You could also delete the `y` variable and use `y_out` throughout the function.

Comment (ppopth) [reply src/c_kzg_4844.c:1102]: I just thought `y_out` is too long. However, I'm fine with either way. Note that `*y_out = y` may not work. I thought about it previously, but it will not work if `fr_t` is not copyable, so I avoided using it.

File: tests/verify_kzg_proof/small/verify_kzg_proof_case_commitment_not_in_G1/data.yaml
Comment (ppopth) [reply tests/verify_kzg_proof/small/verify_kzg_proof_case_commitment_not_in_G1/data.yaml:2]: Yes, I got them from re-runing the test gen from that branch. I also don't know why I got different tests.

**Review by jtraglia — APPROVED** [review]
Nice. LGTM 👍 We need to wait until the spec gets updated before merging though.

File: src/c_kzg_4844.c
Comment (asn-d6) [reply src/c_kzg_4844.c]: > I just thought `y_out` is too long. However, I'm fine with either way. Note that `*y_out = y` may not work. I thought about it previously, but it will not work if `fr_t` is not copyable, so I avoided using it.

Hmm. That's true, but we've been assuming that `fr_t` is copyable in a bunch of places already (check for places where we are copying `FR_ZERO`).

**Review by asn-d6 — APPROVED** [review]
LGTM! Great work!

Let's wait for the spec PR to get merged now.

**Comment by jtraglia** [comment]
The spec has been updated. Merging. Then I'm going to start work on updating the bindings.

**Pull request merged**
